### ISSUE TITLE
Screen reader announces incorrect label for the 'Expand/Collapse (Chevron)' button as 'Toggle the collapsible sidebar category 'Components'.

### ISSUE THREAD

**Issue by msftedad** [facebook/docusaurus#9162]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

Screen reader announces incorrect label for the 'Expand/Collapse (Chevron)' button as 'Toggle the collapsible sidebar category 'Components'.

User is not able to know role of the button.

Screen reader should announce about the group associated with suggestions available while navigating along the suggestions. 
i.e., When user enter the suggestions group, screen reader should announce as '<Group Name>and <suggestion name> along <position set>.

User with visual impairment who rely on screen reader will face difficulties if screen reader announces incorrect information if incorrect label for the 'Expand/Collapse (Chevron)' button.

### Steps to reproduce

**Pre-Requisites:** Scan off mode.

**Repro Steps:**
1.  Open https://www.fast.design/ URL.
2.  Navigate to link ‘Documentation’ and invoke it.
3.  Navigate to 'Expand/Collapse (Chevron)' button and activate it using 'Enter' key and listen to screen reader announcement.

### Expected behavior

Name property should be defined as 'Expand/Collapse Using the Components' and screen reader should announce as 'Expand using the Components' button when expanded and viceversa.

(or) 

Name property can be defined as 'Using the Components' and Aria Expand/collapse state should be defined.

On navigating screen reader should announce as 'Using the Components, button, Expanded' and viceversa. 

### Actual behavior

Screen reader announces incorrect label for the 'Expand/Collapse (Chevron)' button as 'Toggle the collapsible sidebar category 'Components'.

### Your environment

- Public source code: 
- Public site URL: https://www.fast.design/
- Docusaurus version used: Version: 2.4.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Microsoft Edge
                                                                                                       **Version** 114.0.1823.86 (Official build) (64-bit)
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): **OS Name**:  Microsoft Windows 11 Enterprise
                                                                                              **Version**:   10.0.22621 Build 22621

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by pinakipb2** [comment]
@msftedad do check the deploy preview of the above PR and confirm if the issue is resolved. Thanks !

**Comment by slorber** [comment]
@msftedad please tell us if this PR https://github.com/facebook/docusaurus/pull/9269 solves the accessibility problem.

A preview you can test the change on: https://deploy-preview-9269--docusaurus-2.netlify.app/

**Comment by pinakipb2** [comment]
@msftedad Have a look and confirm if the issue is resolved.
Thanks !

**Comment by pinakipb2** [comment]
@msftedad Have a look and confirm if the issue is resolved.
Thanks !

**Comment by msftedad** [comment]
@pinakipb2 @slorber, issue has been resolved. Thanks

**Comment by pinakipb2** [comment]
@msftedad thanks for the confirmation. @slorber can you please merge the PR.
Thanks
